python/ray/services.py

```diff
@@ -450,7 +454,10 @@ def block_sigint():
         stdout=stdout_file,
         stderr=stderr_file,
         stdin=subprocess.PIPE if pipe_stdin else None,
-        preexec_fn=block_sigint)
+        preexec_fn=preexec_fn if sys.platform != "win32" else None)
+
+    if win32_fate_sharing:
```

Comment:
Maybe a comment here saying that it needs to be after spawning the process on windows?

Reply:
That's the only way it can possibly be called though, right? (Given it takes the `Popen` object as an argument.)